cmd/cosign/cli/signingconfig.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2025 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/signingconfig"
+	"github.com/spf13/cobra"
+)
+
+func SigningConfig() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "signing-config",
+		Short: "Interact with a Sigstore protobuf signing config",
+		Long:  "Tool for interacting with a Sigstore protobuf signing config",
+	}
+
+	cmd.AddCommand(signingConfigCreate())
+
+	return cmd
+}
+
+func signingConfigCreate() *cobra.Command {
+	o := &options.SigningConfigCreateOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a Sigstore protobuf signing config",
+		Long: `Create a Sigstore protobuf signing config by supplying verification material for Fulcio, Rekor, OIDC, and TSA services.
+Each service is specified via a repeatable flag (--fulcio, --rekor, --oidc-provider, --tsa) that takes a comma-separated list of key-value pairs.`,
+		Example: `cosign signing-config create \
+    --fulcio="url=https://fulcio.sigstore.dev,api-version=1,start-time=2024-01-01T00:00:00Z,end-time=2025-01-01T00:00:00Z,operator=sigstore.dev" \
```

Comment:
This API is pretty different from the `trusted-root create` API. If we were designing from scratch I see the merits of doing it this way, but since we have a preexisting pattern I think it would be more user-friendly to follow the same pattern.

Reply:
This API allows users to provide multiple service deployments, such as multiple Rekor shards, whereas the `trusted-root create` API allows for only one deployment per service.

I've filed https://github.com/sigstore/cosign/issues/4282 to track copying over this API to `trusted-root create`.

Reply:
https://github.com/sigstore/cosign/pull/4285 for `trusted-root create`. There's a little duplication, using the same `parseKVs` function, I'll update this once we merge one or the other.